### PR TITLE
[MT-4988] Fix - Unable to add link to text in story

### PR DESCRIPTION
**Pull request by e1himself**
- Allow to input incomplete URLs as `href` for `LinkNode` and `ImageNode`
- Normalize incomplete URLs (without `http://` protocol prefix) to always have protocol
- Change `href` validation to only allow URL-containing placeholders (not any placeholder, as it was before)
- Refactor URL detecting RegExp code -- move it to the top-level `lib` folder
- Drop unused `LinkMenu` module -- it has been replaced by it's reincarnation in `rich-formatting-menu` extension long ago
- Fix `updateImage` to update the current image node disregarding the selection. Otherwise an update triggered by putting focus somewhere else was lost.

### CODE REVIEW

File: packages/slate-editor/src/extensions/image/components/ImageMenu.tsx [hunk 1]
```diff
@@ -136,15 +135,17 @@ export function ImageMenu({
         function (href: string, valid: boolean) {
             setHref(href);
             if (valid) {
-                onChange({ href });
+                onChange({ href: normalizeHref(href) });
```
Comment (e1himself) [inline packages/slate-editor/src/extensions/image/components/ImageMenu.tsx:138]: This is an actual fix for MT-4988

File: packages/slate-editor/src/modules/rich-formatting-menu/LinkMenu.tsx [hunk 2]
```diff
@@ -31,7 +30,7 @@ export function LinkMenu({
     const [new_tab, setNewTab] = useState(node?.new_tab ?? true);
 
     function handleSave() {
-        onChange({ href, new_tab });
+        onChange({ href: normalizeHref(href), new_tab });
```
Comment (e1himself) [inline packages/slate-editor/src/modules/rich-formatting-menu/LinkMenu.tsx:33]: This is an actual fix for MT-4988

File: packages/slate-editor/src/modules/rich-formatting-menu/LinkMenu.tsx [hunk 3]
```diff
@@ -58,9 +57,9 @@ export function LinkMenu({
                                             value={href}
                                             onChange={setHref}
                                             icon={Link}
-                                            pattern={STRING_URL_PATTERN}
+                                            pattern={HREF_REGEXP.source}
                                             placeholder="Paste link"
-                                            type="url"
```
Comment (e1himself) [inline packages/slate-editor/src/modules/rich-formatting-menu/LinkMenu.tsx:63]: This is an actual fix for MT-4988

**Review by aspirisen — APPROVED** [review]

**Pull request merged**